plugins/inputs/logfile/tail/tail.go

```diff
@@ -309,6 +311,7 @@ func (tail *Tail) readlineUtf16() (string, error) {
 func (tail *Tail) tailFileSync() {
 	defer tail.Done()
 	defer tail.close()
+	defer tail.TailerQueue.Dequeue()
```

Comment:
Not sure why we need to include another `defer` statement here. If dequeuing from this structure is part of cleanup, it should just be put in the `Done` or `close` function rather than adding an additional defer here.